ntp-proto/src/peer.rs

```diff
@@ -219,19 +217,19 @@ impl Peer {
         );
 
         match updated {
-            None => MsgForSystem::NoUpdate,
+            None => Err(PeerError::InvalidPacketTime),
```

Comment:
This is not really the correct condition to give back, as it isn't that we got packets out of order here, just that the most recent packet isn't (yet) good enough to consider for the rest of the algorithms.